File: Java/src/TransposeMatrixTest.java
import org.junit.Assert;
import org.junit.Test;

public class TransposeMatrixTest {

    @Test
    public void testTranspose() {
        TransposeMatrix tm = new TransposeMatrix();

        Assert.assertEquals(tm.transpose(new int[][]{{1, 1}}), new int[][]{{1}, {1}});

        Assert.assertEquals(tm.transpose(new int[][]{{1}, {1}}), new int[][]{{1, 1}});

        Assert.assertEquals(tm.transpose(new int[][]{{1, 2, 3}, {4, 5, 6}, {7, 8, 9}}), new int[][]{{1, 4, 7}, {2, 5, 8}, {3, 6, 9}});

        Assert.assertEquals(tm.transpose(new int[][]{{1, 2, 3}, {4, 5, 6}}), new int[][]{{1, 4}, {2, 5}, {3, 6}});
    }

}

File: Kotlin/src/TransposeMatrixTest.kt
import org.junit.Assert
import org.junit.Test

class TransposeMatrixTest {

    @Test
    fun testTranspose() {
        val tm = TransposeMatrix()

        Assert.assertEquals(tm.transpose(arrayOf(intArrayOf(1, 1))), arrayOf(intArrayOf(1), intArrayOf(1)))

        Assert.assertEquals(tm.transpose(arrayOf(intArrayOf(1), intArrayOf(1))), arrayOf(intArrayOf(1, 1)))

        Assert.assertEquals(tm.transpose(arrayOf(intArrayOf(1, 2, 3), intArrayOf(4, 5, 6), intArrayOf(7, 8, 9))), arrayOf(intArrayOf(1, 4, 7), intArrayOf(2, 5, 8), intArrayOf(3, 6, 9)))

        Assert.assertEquals(tm.transpose(arrayOf(intArrayOf(1, 2, 3), intArrayOf(4, 5, 6))), arrayOf(intArrayOf(1, 4), intArrayOf(2, 5), intArrayOf(3, 6)))
    }

}
